### PR TITLE
Add the snapshot configuration field

### PR DESCRIPTION
**Pull request by voscob**
### Description

In this pull request, a new configuration boolean field `snapshot` has been added, determining whether the connector will take a snapshot of the entire table before starting CDC mode. The default value is `true`.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-oracle/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: source/iterator/iterator.go [hunk 1]
```diff
@@ -82,8 +118,16 @@ type Params struct {
 func New(ctx context.Context, params Params) (*Iterator, error) {
 	var err error
 
+	// hash the table name to use it as a postfix in the tracking table and snapshot,
+	// because the maximum length of names (tables, triggers, etc.) is 30 characters
+	h := fnv.New32a()
```
[the remark this answers is not in the document]
Comment (voscob) [reply source/iterator/iterator.go:123]: No, I just moved it from the `/source/iterator/snapshot.go`.

File: README.md [hunk 2]
```diff
@@ -97,6 +97,7 @@ The `last_processed_val` field represents the last processed element value, and
...
 | `keyColumns`     | Comma-separated list of column names to build the `sdk.Record.Key`. See more: [key handling](#key-handling).                                                        | false    | `id,name`                                   |
+| `snapshot`       | Whether the connector will take a snapshot of the entire table before starting cdc mode. The default is `true`.                                                     | false    | `false`                                     |
 | `columns`        | List of column names that should be included in each Record's payload, by default includes all columns.                                                             | false    | `id,name,age`                               |
 | `batchSize`      | Size of rows batch. Min is 1 and max is 100000. The default is 1000.                                                                                                | false    | `100`                                       |
```
[the remark this answers is not in the document]
Comment (voscob) [reply README.md:105]: Done.

File: README.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ WHERE {{keyColumn}} > {{position.last_processed_val}}
 FETCH NEXT {{batchSize}} ROWS ONLY;
 ```
 
+This behavior is enabled by default, but can be turned off by adding `"snapshot": "false"` to the Source configuration.
```
Comment (maksenius) [inline README.md:49]: maybe it is must be the last line on this part.
What do you think about it

Comment (voscob) [reply README.md:49]: I thought about it but decided to put it in the last line before the CDC information.
But if you think it's better, I can update.

Comment (maksenius) [reply README.md:49]: lets keep like this

**Review by maksenius — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -46,6 +46,8 @@ WHERE {{keyColumn}} > {{position.last_processed_val}}
...
+
 When all records have been returned, the snapshot is deleted and the connector switches to the CDC mode.
```
Comment (maha-hajja) [inline README.md:51]: what do you mean by the snapshot is deleted?

**Review by maha-hajja — APPROVED** [review]
lgtm

**Pull request merged**